app/models/resource.rb

```diff
@@ -105,6 +104,14 @@ def search account: nil, kind: nil, owner: nil, offset: nil, limit: nil, search:
       scope = scope.textsearch(search) if search
 
       if offset || limit
+        # 'limit' must be an integer greater than 0 if given
+        if limit && (!numeric?(limit) || limit.to_i <= 0)
+          raise ApplicationController::UnprocessableEntity, "'limit' contains an invalid value. 'limit' must be a positive integer."
+        end
+        # 'offset' must be an integer greater than or equal to 0 if given
+        if offset && (!numeric?(offset) || offset.to_i.negative?)
+          raise ApplicationController::UnprocessableEntity, "'offset' contains an invalid value. 'offset' must be an integer greater than or equal to 0."
+        end
```

Comment:
`UnprocessableEntity` is an http level concern.  That is, we should only be talking about that error inside of controllers, and specifically in code that is mapping an actual domain error into an http status code.

Inside of the model we should be raising a more specific validation error, with a name that reflects what went wrong.  `InvalidQueryParameter` or something like that.

Separate comment: There's a lot of formatting / clean changes here (which is awesome), but please put those into a separate commit with a subject like "Cleanup formatting" -- otherwise the "meat" of this PR (which is actually pretty small and simple) isn't clear at a glance.

Reply:
We handle the `UnprocessableEntity` with rendering the response. I think this a redundant addition.

Reply:
About the formatting I agree. But I think that since it is a very small PR (both logic and formatting) the separation here is not mandatory.
Is it a blocker? @jonahx

Reply:
@jonahx fixed